v3/generator/operations/operations.go

```diff
@@ -0,0 +1,694 @@
+	doc := op.Description
+	if doc == "" {
+		doc = op.Summary
+	}
+	if doc == "null" {
+		return ""
+	}
+
+	docs := strings.Split(doc, "\n")
+	r := []string{}
+	for i, d := range docs {
+		if d == "" {
+			docs = append(docs[:i], docs[i+1:]...)
+			continue
+		}
+		r = append(r, "// "+strings.TrimSpace(d))
+	}
+
+	if len(r) == 0 {
+		return ""
+	}
+
+	return strings.Join(r, "\n")
+}
+
+func renderURLPathBuilder(rawPath string, op *v3.Operation) string {
+	if len(op.Parameters) == 0 {
+		return fmt.Sprintf("%q", rawPath)
+	}
+	path := rawPath
+	sprintfParam := []string{}
+	for _, p := range op.Parameters {
+		if p.In != "path" {
+			continue
+		}
+		path = strings.Replace(path, "{"+p.Name+"}", "%v", 1)
+		sprintfParam = append(sprintfParam, helpers.ToLowerCamel(p.Name))
+	}
+	if path == rawPath {
+		return fmt.Sprintf("%q", rawPath)
+	}
+
+	return `fmt.Sprintf("` + path + `", ` + strings.Join(sprintfParam, ", ") + ")"
+}
+
+func getQueryParams(op *v3.Operation) map[string]string {
+	if len(op.Parameters) == 0 {
+		return nil
+	}
+
+	result := make(map[string]string)
+	for _, p := range op.Parameters {
+		if p.In != "query" {
+			continue
+		}
+		result[p.Name] = helpers.ToLowerCamel(p.Name)
+	}
+	if len(result) == 0 {
+		return nil
+	}
+
+	return result
+}
+
+type Method struct {
+	Comment        string
+	APIName        string
+	Name           string
+	Params         string
+	ValueReturn    string
+	URLPathBuilder string
+	HTTPMethod     string
+	BodyRequest    bool
+	BodyRespType   string
+	ContentType    string
+	QueryParams    map[string]string
+}
+
+const methodTemplate = `
+{{ .Comment }}
+func (c Client{{ .APIName }}) {{ .Name }}({{ .Params }}) {{ .ValueReturn }} {
+	path := {{ .URLPathBuilder }}
+
+	u, err := url.Parse(c.serverURL + path)
+	if err != nil {
+		return nil, err
+	}
+
+	{{ if .BodyRequest }}
+	body, err := prepareJsonBody(req)
+	if err != nil {
+		return nil, err
+	}
+	{{ end }}
+
+	request, err := http.NewRequestWithContext(c.ctx, "{{ .HTTPMethod }}", u.String(), {{ if .BodyRequest }}body{{else}}nil{{end}})
+	if err != nil {
+		return nil, err
+	}
+	request = request.WithContext(c.ctx)
+
+	{{ if ne .QueryParams nil }}if len(opts) > 0 {
+		q := request.URL.Query()
+		for _, opt := range opts {
+			opt(q)
+		}
+		request.URL.RawQuery = q.Encode()
+	}{{ end }}
+
+	{{ if ne .ContentType "" }}
+	request.Header.Add("Content-Type", "{{ .ContentType }}")
+	{{ end }}
+
+	if err := registerRequestMiddlewares(&c, c.ctx, request); err != nil {
+		return nil, err
+	}
+
+	resp, err := c.httpClient.Do(request)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode < http.StatusOK || resp.StatusCode > http.StatusIMUsed {
+		return nil, fmt.Errorf("request returned %d code", resp.StatusCode)
+	}
+
+	{{ if ne .ValueReturn "error" }}
+	bodyresp := {{ .BodyRespType }}{}
+	if err := prepareJsonResponse(resp, bodyresp); err != nil {
+		return nil, err
+	}
+
+	return bodyresp, nil{{ else }}return nil{{end}}
```

Comment:
Do we really have some endpoints that don't return a value but only an error ?

Reply:
I removed it and left a reminder in case.